src/multiset.rs

```diff
@@ -108,6 +108,29 @@ where
         }
     }
 
+    /// An iterator visiting all elements in arbitrary order paired with the number of times the
+    /// item appears.
+    /// The iterator element type is `(&'a K, usize)`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use multiset::HashMultiSet;
+    /// let mut multiset = HashMultiSet::new();
+    /// multiset.insert(0);
+    /// multiset.insert(0);
+    /// multiset.insert(1);
+    ///
+    /// let mut keys_with_counts: Vec<_> = multiset.iter_counts().collect();
+    /// keys_with_counts.sort(); // Order is arbitrary
+    /// assert_eq!(keys_with_counts, vec![(&0, 2), (&1, 1)]);
+    /// ```
+    pub fn iter_counts(&self) -> IterCounts<K> {
```

Comment:
This can probably be done with less code. Something like this should work:
```
pub fn iter_counts(&self) -> impl Iterator<Item = (&K, usize)> {
    self.elem_counts.iter().map(|(key, &counts)| (key, counts))
```

Reply:
It's true. The general practice for the standard library has been to provide a concrete struct to return. However, returning an `impl` trait is a relatively recent change, so that may be considered a new best practice. I'd be willing to go that way. It may be worth considering for the sake of consistency, since we do need custom structs for several of the other methods (i.e. `UnionCounts`, etc.).